File: src/main/java/utils/LinkedListReversal.java
package utils;

public class LinkedListReversal<T> {

    public LinkedListNode<T> reverseLinkedList(LinkedListNode<T> node) {
        LinkedListNode<T> prev = null;
        LinkedListNode<T> current = node;
        LinkedListNode<T> next;

        while (current != null) {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }
        node = prev;
        return node;
    }

}

File: src/main/kotlin/utilskotlin/LinkedListReversal.kt
package utilskotlin

class LinkedListReversal<T> {

    fun reversedLinkedListNode(node: LinkedListNode<T>?): LinkedListNode<T>? {

        var current: LinkedListNode<T>? = node
        var prev: LinkedListNode<T>? = null
        var next: LinkedListNode<T>?

        while(current != null) {
            next = current.next
            current.next = prev
            prev = current
            current = next
        }

        return prev
    }

}
